client/src/main/java/com/paypal/selion/platform/grid/SeleniumGridListener.java

```diff
@@ -125,6 +126,8 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
             }
         } catch (Exception e) { // NOSONAR
             logger.log(Level.WARNING, "An error occurred while processing beforeInvocation: " + e.getMessage(), e);
+            //Failing the test case, because pre validation/configuration got failed
+            Assert.fail(e.getMessage(), e);
```

Comment:
@sundaramrajendran I'm in agreement with @krmahadevan.
The primary issue with this approach is the failure message will be associated with the class `SeleniumGridListener`

Is there a more elegant way of doing this? What happens if you re-throw the `Exception` to TestNG?